caikit/core/exceptions/caikit_core_exception.py

```diff
@@ -0,0 +1,36 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Caikit Core Exception enum used for reporting Exception status raised in caikit core
+"""
+
+# Standard
+from enum import Enum
+
+
+class CaikitCoreStatusCode(Enum):
+    NOT_FOUND = 1
+    INVALID_ARGUMENT = 2
+    CONNECTION = 3
```

Comment:
I might say `CONNECTION_ERROR` or `CONNECTION_LOST` or something to indicate that the connection is the problem, what do you think?